### DISCUSSION TITLE
Packaging clrjit.dll in a Single File Executable

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#107377] [Q&A]
I have a single-file app that requires clrjit be packaged with it.

When I build in debug, I can see the file in the ```win-x64``` directory but when I publish with single-file, the DLL is nowhere to be found in the ```%temp%\.net``` folder.  It seems like it is not packaged at all.

How do I ensure this file is packaged?

Here is my pubxml:
```
<?xml version="1.0" encoding="utf-8"?>
<!--
https://go.microsoft.com/fwlink/?LinkID=208121.
-->
<Project>
	<PropertyGroup>
		<Configuration>Release</Configuration>
		<Platform>Any CPU</Platform>
		<PublishDir>bin\Release\net8.0\publish\win-x64\</PublishDir>
		<PublishProtocol>FileSystem</PublishProtocol>
		<_TargetId>Folder</_TargetId>
		<TargetFramework>net8.0</TargetFramework>
		<RuntimeIdentifier>win-x64</RuntimeIdentifier>

		<SelfContained>true</SelfContained>
		<PublishSingleFile>true</PublishSingleFile>
		<PublishReadyToRun>false</PublishReadyToRun>
		<PublishTrimmed>false</PublishTrimmed>

		<IncludeNativeLibrariesForSelfExtract>true</IncludeNativeLibrariesForSelfExtract>
		<EnableCompressionInSingleFile>true</EnableCompressionInSingleFile>


	</PropertyGroup>
</Project>
```

Thanks so much for the help!

**Comment by vitek-karas** [comment]
For single-file the JIT is statically linked into the executable. Simple console app should produce just one executable with single-file, no external dlls. The way this is done is that the prebuilt single-file exe already includes the JIT (and the rest of the runtime) and SDK only adds managed code into it.

Does your app work as single-file, or is there some other reason you're looking into this?

**Reply by TonyValenti** [reply]
Hi @vitek-karas -
Thank you for the explanation. 

Our app does not work when published as a single file.

**Reply by vitek-karas** [reply]
What is the failure you  get?

**Reply by TonyValenti** [reply]
We use Harmony to do runtime method patching of certain assemblies that our application uses. The Harmony and MonoMod libraries cannot work with single file executable because of how the DLL is baked in to the app host.

**Reply by vitek-karas** [reply]
Thanks for the explanation. Not sure how I can help here - there are definitely features and functionality single-file is not compatible with. And some external tools as well. If that's the case I'm afraid the only answer is to not use single-file.

**Reply by jkotas** [reply]
Note that Harmony and similar libraries modify runtime internals in unsupported way and they are known to cause reliability issues (e.g. #83034). These libraries are not suitable for use in production.
